drivers/interrupt_controller/CMakeLists.txt

```diff
@@ -44,6 +44,7 @@ zephyr_library_sources_ifdef(CONFIG_NXP_PINT                intc_nxp_pint.c)
 zephyr_library_sources_ifdef(CONFIG_RENESAS_RX_ICU          intc_renesas_rx_icu.c)
 zephyr_library_sources_ifdef(CONFIG_RENESAS_RZ_EXT_IRQ      intc_renesas_rz_ext_irq.c)
 zephyr_library_sources_ifdef(CONFIG_NXP_IRQSTEER            intc_nxp_irqsteer.c)
+zephyr_library_sources_ifdef(CONFIG_NXP_IRQSTEER1           intc_nxp_irqsteer1.c)
```

Comment:
Please explain why do you need a new driver? 
Why the `intc_nxp_irqsteer.c` driver is not enough?

In the commit message you state that the "old" `intc_nxp_irqsteer` driver can be replaced.
In this case, why don't you enhance the `intc_nxp_irqsteer` driver with the support that you need for i.MX943?

Reply:
> @biwenli so your proposal is to replace the "old" irqsteer driver with the new one?
> 
> Usually, when such a thing is happening you need to update all the references to the old driver with the new one. And as last point remove the old driver.

Yes, the old driver will be phased out over time. New product will use the new driver directly.

Reply:
@biwenli Could you please add a note indicating that the old `irqsteer` will be deprecated and should no longer be used in new samples or products?
Also, it would be helpful to include at least one commit where the old `irqsteer` is replaced with the new implementation, so others can refer to it as an example.
Thanks.

Reply:
> @biwenli Could you please add a note indicating that the old `irqsteer` will be deprecated and should no longer be used in new samples or products? Also, it would be helpful to include at least one commit where the old `irqsteer` is replaced with the new implementation, so others can refer to it as an example. Thanks.

@iuliana-prodan ,

Sure, np. Thanks.